src/ProxyServer/ProxyServer.swift

```diff
@@ -29,6 +29,8 @@ open class ProxyServer: NSObject, TunnelDelegate {
     open var observer: Observer<ProxyServerEvent>?
 
     var tunnels: TunnelArray = []
+    
+    private let accessQueue = DispatchQueue(label: "NEKit.ProxyServer.TunnelArrayAccess", attributes: .concurrent)
```

Comment:
No. All blocks are executed concurrently. `sync` only guarantees that it will continue after the block is executed.

Plus, there should be only one dispatch queue running, so there is no race condition. That's why though `Atomic` exists but is no longer used.